WebAssembly.Tests/TestUtility.cs

```diff
@@ -86,4 +86,23 @@ private static void AreNotEqual(object? a, object? b)
         if (b != null)
             Assert.IsNotNull(b.ToString());
     }
+
+    /// <summary>
+    /// Tests that an exception is thrown with the expected type and message.
+    /// </summary>
+    public static void ThrowsException<T>(string message, Action action)
+        where T : Exception
+    {
+        var baseMessage = $"Expected {typeof(T).FullName} with message '{message}'";
+        try
+        {
+            action();
+            Assert.Fail($"{baseMessage} but no exception was thrown");
+        }
+        catch (Exception e)
+        {
+            if (e.GetType() != typeof(T) || e.Message != message)
```

Comment:
As noted above, I think this whole helper can be removed, but for this specific thing I C# 6 introduced the `when` keyword:
```suggestion
        catch (Exception e) when (e.GetType() != typeof(T) || e.Message != message)
        {
```

Reply:
Ah yes good call. Old habits die hard